stl/inc/format

```diff
@@ -2793,6 +2918,25 @@ template <_Format_supported_charT _CharT, class _Traits>
 struct formatter<basic_string_view<_CharT, _Traits>, _CharT>
     : _Formatter_base<basic_string_view<_CharT, _Traits>, _CharT, _Basic_format_arg_type::_String_type> {};
 
+template <class _CharT, class... _Args>
+struct _Basic_format_string {
+    basic_string_view<_CharT> _Str;
+
+    template <class _Ty>
+        requires convertible_to<const _Ty&, basic_string_view<_CharT>>
+    consteval _Basic_format_string(const _Ty& _Str_val) : _Str(_Str_val) {
+        if (_Is_execution_charset_utf8()) {
+            _Parse_format_string(_Str, _Format_checker<_CharT, remove_cvref_t<_Args>...>{_Str});

```

Comment:
Doesn't stripping cv-qualifiers here mean we won't notice if an arg type is `const` but not `const`-formattable? (Suggests I am confused, or a lack of test coverage.)

Reply:
^^ This is the real comment - the rest of these are "only if we need to touch something else."

Reply:
we don't miss that case because `_Format_arg_traits::_Storage_type` adds it back, and, in any case, all the args are passed by `const Args&...` in this PR anyway.

Pre P2418 (which this PR is) formatters kinda had to accept `const T&` to work at all, if you provide a non-const format we never call it, even with non-const arguments (since the const is added when you call format/make_format_args).